addons/sms/static/src/messaging/component/message/message.js

```diff
@@ -0,0 +1,32 @@
+odoo.define('sms.messaging.component.Message', function (require) {
+'use strict';
+
+const components = {
+    Message: require('mail.messaging.component.Message'),
+};
+
+const { patch } = require('web.utils');
+
+patch(components.Message, 'sms.messaging.component.Message', {
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _onClickFailure() {
+        if (this.message.message_type === 'sms') {
+            this.env.do_action('sms.sms_resend_action', {
+                additional_context: {
+                    default_mail_message_id: this.message.id,
+                },
+            });
+        } else {
+            this._super(...arguments);
+        }
+    },
```

Comment:
feels like this could/should be moved into entity too?

Reply:
Yes, IMO components should just care for rendering and having intercepts of user interactions (e.g. click, keydown, etc.), and the actual handling of intercepts should be in entity.

Reply:
done